pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManagerTest.java

```diff
@@ -639,6 +643,29 @@ public void testBatchUpload()
     Assert.assertEquals(segmentLineage.getLineageEntry(lineageEntryId2).getState(), LineageEntryState.COMPLETED);
   }
 
+  private void waitForEVUpdateForSegments(String tableNameWithType, Set<String> segments)
```

Comment:
The API should handle this wait instead of the client. User doesn't need to understand that they need to wait for external view to change before ending the replacement

Reply:
changed the API to wait online segments for some time.